src/mdacli/cli.py

```diff
@@ -271,147 +291,258 @@ def create_CLI(cli_parser, interface_name, parameters):
 
-    # Convert special types (i.e AtomGroups)
-    # Ugly that we have to parse again... but currently I have no better idea :(
-    params = parse_docs(analysis_callable)[2]  # Index [2] for paramaters
-    for param_name, dictionary in params.items():
-        if "AtomGroup" in dictionary['type']:
-            sel = u.select_atoms(kwargs[param_name])
-            if len(sel) > 0:
-                kwargs[param_name] = sel
-            else:
-                raise ValueError(
-                    "AtomGroup `-{}` with selection `{}` does not "
-                    "contain any atoms".format(
-                        param_name,
-                        kwargs[param_name]
-                        )
-                    )
-        elif "Universe" in dictionary['type']:
-            kwargs[param_name] = u
-
-    with warnings.catch_warnings():
-        warnings.simplefilter('always')
-        startframe = convert_str_time(kwargs.pop("begin"), u.trajectory.dt)  # noqa: E501
-        stopframe = convert_str_time(kwargs.pop("end"), u.trajectory.dt)  # noqa: E501
-        step = convert_str_time(kwargs.pop("dt"), u.trajectory.dt)
-
-        # raises error if frame selection range is an empty selection
-        if not list(range(u.trajectory.n_frames)[slice(startframe, stopframe, step)]):  # noqa: E501
-            raise ValueError("Trajectory frame range {}:{}:{} is not valid for {} frames."  # noqa: E501
-                             "".format(startframe, stopframe, step, u.trajectory.n_frames))  # noqa: E501
-
-    # Collect paramaters not necessary for initilizing ac object.
-    output_directory = kwargs.pop("output_directory")
-    output_prefix = kwargs.pop("output_prefix")
-    output_prefix += "_" if len(output_prefix) > 0 else ""
-
-    ac = analysis_callable(**kwargs)
-    ac.run(start=startframe,
-           stop=stopframe,
-           step=step,
-           verbose=verbose)
+    # Run the analysis
+    for key, value in run_parameters.items():
+        run_parameters[key] = convert_str_time(value, universe.trajectory.dt)
 
+    ac.run(verbose=verbose, **run_parameters)
+
+    # Save results
     try:
         ac.save_results()
 
     except AttributeError:
-        save_results(
-            ac.results,
-            os.path.join(
-                output_directory,
-                f"{output_prefix}{type(ac).__name__}"
-                ),
-            )
+        direcory = ""
+        try:
+            direcory += output_parameters["output_directory"]
+        except KeyError:
+            pass
+
+        fname = ""
+        try:
+            fname += output_parameters["output_prefix"] + "_"
+        except KeyError:
+            pass
+        fname += analysis_callable.__name__
+        save_results(ac.results, os.path.join(direcory, fname))
+
     return ac
 
 
-def maincli(ap):
-    """Execute main client interface."""
-    if len(sys.argv) < 2:
-        ap.error("A subcommand is required.")
+def convert_analysis_parameters(universe,
```

Comment:
This looks nice. Help me understand.

This function addresses the `Universe` parameter that some Analysis Classes require. This Universe parameter is being identified as "mandatory" in the CLI though. For example:

```bash
mdacli AlignTraj -h
```

But we can run nicely the following command:

```
mdacli AlignTraj -s data/topol.tpr -f data/traj.trr -reference "name CA"
```

Do you think we should remove those mandatory arguments? I believe I am confused here. My fault :face_with_head_bandage:

Reply:
No you are right. The Universe parameter is a weird thing. You can type whatever you want for `mobile` or `reference` but we always set these two to the universe created from the `-s` and `-f` option. We also have to to this since there is no other universe...